### PR TITLE
chore: refactor useraccounts to use api layer

### PR DESCRIPTION
**Pull request by subirjolly**
Related: #5029

Refactor user accounts page (UserAccount.tsx) so that `getAccounts` and `patchAccount` from unityroutes are being used by the auth API layer.

### Checklist

Authors and Reviewer(s), please verify the following:

- [x] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [x] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [x] Documentation updated or issue created (provide link to issue/PR)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [ ] Feature flagged, if applicable

### CODE REVIEW

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx:102]: good catch!

Comment (subirjolly) [reply src/accounts/context/userAccount.tsx:102]: Double checked that only 1 account can be `isActive` at any time. So, I will make this change.

Comment (subirjolly) [reply src/accounts/context/userAccount.tsx]: updated!

File: src/accounts/context/userAccount.tsx
Comment (wdoconnell) [inline src/accounts/context/userAccount.tsx:92]: This isn't absolutely necessary, but one suggestion I have is that `line` could also be more semantic - it might be clearer if the element were `acct` or `account`.

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx]: updated

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx]: updated

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx]: deliberate usage of `map` here so that I am not modifying the state variable directly. refactored other parts.

File: src/identity/apis/auth.ts
Comment (hoorayimhelping) [inline src/identity/apis/auth.ts:340]: this can throw `401`, `422` and `500`, so let's handle those errors similar to how the other functions in this file handle errors

File: src/accounts/context/userAccount.tsx
Comment (wdoconnell) [inline src/accounts/context/userAccount.tsx:99]: Same here re: `line`

File: src/accounts/context/userAccount.tsx
Comment (wdoconnell) [inline src/accounts/context/userAccount.tsx:111]: I don't see `dispatch` or `defaultAccountId` in here - was this a mistake, or were these present before?

Is there anything else missing from the useEffect dependency array that would create a new warning?

File: src/accounts/context/userAccount.tsx
Comment (wdoconnell) [inline src/accounts/context/userAccount.tsx:131]: What's the rationale for adding a `useCallback` here?

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx]: i was just moving the existing code. but good catch. i will rename it.

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx:111]: they were extraneous. my changes seem to be working fine but i will test my changes again.

File: src/identity/apis/auth.ts
Comment (subirjolly) [reply src/identity/apis/auth.ts]: updated

File: src/accounts/context/userAccount.tsx
Comment (wdoconnell) [reply src/accounts/context/userAccount.tsx]: This is probably okay either way, as the old code set the `.name` property directly. But I'm also a little leery of directly manipulating context even if we're about to call setUserAccounts anyway. The O(n) from `map` isn't going to be expensive in a meaningful way, and following the usual pattern of not manipulating state directly keeps us out of trouble if we have to change this later. 

For example, we'll probably move the account state into redux eventually. And in redux we'll be foreclosed from making direct changes to state because the state will be frozen by immer.

Comment (wdoconnell) [reply src/accounts/context/userAccount.tsx]: @hoorayimhelping Do you have a view on this one?

Comment (hoorayimhelping) [reply src/accounts/context/userAccount.tsx]: The code here has changed since these comments were made, so I'm trying to get the context of the original discussion, but even without a ton of context, there are two things to consider here:

1. If we're thinking about the performance of this, keep in mind that using the spread operator is not a free operation. It likely has the same or slightly higher cost as a `O(n)` `map` operation, since it needs to iterate through all the elements of a collection and copy them (the copying operation accounting for the slightly higher cost)
2. All things (like performance and security) being equal, the best solution is the one that finds the balance between clarity and simplicity. If mapping over an array and changing a single item is necessary to call attention to, do it. If we can do the same thing without a loop, that is probably better

**Comment by wdoconnell** [comment]
FYI - I'll be able to review first thing tomorrow.

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx]: updated!

File: src/identity/apis/auth.ts
Comment (wdoconnell) [inline src/identity/apis/auth.ts:347]: What's the rationale for this one - do we expect anything other than a `200`, `422`, or `500`? This does look more cautious, but I only see `200`, `422`, and the `default` server error in the openapi spec.

**Review by wdoconnell — CHANGES_REQUESTED** [review]
I had a few more questions, but otherwise this looks good to me - I might add the video showing this working to the PR too.

File: src/identity/apis/auth.ts
Comment (subirjolly) [reply src/identity/apis/auth.ts:347]: yea, being extra cautious. makes sense to remove it. will do.

**Comment by subirjolly** [comment]
WORKS

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx:151]: will clean this up as part of https://github.com/influxdata/ui/issues/5802

**Comment by subirjolly** [comment]
> I had a few more questions, but otherwise this looks good to me - I might add the video showing this working to the PR too.

updated!

File: src/identity/apis/auth.ts
Comment (wdoconnell) [reply src/identity/apis/auth.ts]: Paranoia is good in my book - I think it's okay if you want to leave it in.

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx:135]: absolutely good point. let me check if ever a user can who has a single account can have `inActive` status for that account. if so, then i will add this: `dispatch(notify(defaultAccountNotFoundError()))`. If not then, i will remove this check. Sounds good?

Comment (subirjolly) [reply src/accounts/context/userAccount.tsx:135]: I confirmed that it's not possible to have 0 account(s) which have `isActive` as `false`. So, removing this check.

You guys ok with this? ☝️  @asalem1 @wdoconnell

Comment (subirjolly) [reply src/accounts/context/userAccount.tsx]: done!

Comment (wdoconnell) [reply src/accounts/context/userAccount.tsx]: Sounds good to me too

Comment (wdoconnell) [reply src/accounts/context/userAccount.tsx]: Since it doesn't have a chance of occurring, I don't think we need the new notification code

**Comment by subirjolly** [comment]
Resolved merge commits. Require approval again 🙏

File: src/accounts/context/userAccount.tsx
Comment (wdoconnell) [inline src/accounts/context/userAccount.tsx:97]: I don't think hold up is necessary here, but do we want this to go to honeybadger where our other error reporting goes?  

Eventing will just go to snowflake, and I don't think the analytics folks are going to do anything with that.

File: src/shared/copy/notifications/categories/accounts-users-orgs.ts
Comment (wdoconnell) [inline src/shared/copy/notifications/categories/accounts-users-orgs.ts:7]: I don't think this notification that's been added is being used anywhere, but please LMK if that's wrong.

**Review by wdoconnell — CHANGES_REQUESTED** [review]
Two more questions on this below

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx:97]: i will remove it in the new PR: https://github.com/influxdata/ui/pull/5810

Comment (subirjolly) [reply src/accounts/context/userAccount.tsx:97]: actually, where else should we report? i think we should report it to somewhere.

File: src/shared/copy/notifications/categories/accounts-users-orgs.ts
Comment (subirjolly) [reply src/shared/copy/notifications/categories/accounts-users-orgs.ts:7]: ah yea, i added this and then decided to go against it. will add it in the above mentioned PR.

Comment (subirjolly) [reply src/shared/copy/notifications/categories/accounts-users-orgs.ts:7]: removed in the other PR.

File: src/accounts/context/userAccount.tsx
Comment (subirjolly) [reply src/accounts/context/userAccount.tsx:97]: Let me use this: `reportErrorThroughHoneyBadger`

File: src/accounts/context/userAccount.tsx
Comment (wdoconnell) [inline src/accounts/context/userAccount.tsx:151]: Pending the tweak to this `reportErrorThroughHoneyBadger` call - adding a more descriptive name, and adding relevant information about the account to context (account name, account id, user name, or id), this looks good to me.

**Review by hoorayimhelping — APPROVED** [review]

**Review by wdoconnell — APPROVED** [review]

**Pull request merged**
